pkg/core/object/replicate.go

```diff
@@ -9,41 +9,45 @@ import (
 )
 
 const (
-	currentVersion = 7 // it is also a number of fields
-)
-
-const (
-	networkMagicKey = "network"
+	// required fields.
 	cidKey          = "cid"
 	oidKey          = "oid"
 	sizeKey         = "size"
-	deletedKey      = "deleted"
-	lockedKey       = "locked"
 	validUntilKey   = "validUntil"
+	networkMagicKey = "network"
+
+	// optional fields.
+	deletedKey = "deleted"
+	lockedKey  = "locked"
 )
 
 // EncodeReplicationMetaInfo uses NEO's map (strict order) serialized format as a raw
 // representation of object's meta information.
 //
 // This (ordered) format is used (keys are strings):
 //
-//	"network": network magic
 //	"cid": _raw_ container ID (32 bytes)
 //	"oid": _raw_ object ID (32 bytes)
 //	"size": payload size
-//	"deleted": array of _raw_ object IDs
-//	"locked": array of _raw_ object IDs
 //	"validUntil": last valid block number for meta information
+//	"network": network magic
+//	"deleted": [OPTIONAL] array of _raw_ object IDs
+//	"locked": [OPTIONAL] array of _raw_ object IDs
 func EncodeReplicationMetaInfo(cID cid.ID, oID oid.ID, pSize uint64,
 	deleted, locked []oid.ID, vub uint64, magicNumber uint32) []byte {
 	kvs := []stackitem.MapElement{
-		kv(networkMagicKey, magicNumber),
 		kv(cidKey, cID[:]),
 		kv(oidKey, oID[:]),
 		kv(sizeKey, pSize),
-		oidsKV(deletedKey, deleted),
-		oidsKV(lockedKey, locked),
 		kv(validUntilKey, vub),
+		kv(networkMagicKey, magicNumber),
+	}
+
+	if len(deleted) > 0 {
```

Comment:
i'd split unit tests into `min` and `full` cases. And include fields added in the following commits

Reply:
done